### PR TITLE
propagate cancellation tokens

### PR DESCRIPTION
**Pull request by jmarolf**
I am just a simple developer trying to send these cancellation tokens downstream where they belong

### CODE REVIEW

**Comment by mavasani** [comment]
Also tagging @carlossanlop, who is working on analyzer/fixer for this in https://github.com/dotnet/roslyn-analyzers/pull/3641.

File: src/Compilers/VisualBasic/Portable/Symbols/Source/SourceModuleSymbol.vb [hunk 1]
```diff
@@ -703,7 +703,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                                 Sub()
                                     Try
                                         visitor(symbol)
-                                    Catch e As Exception When FatalError.ReportUnlessCanceled(e)
+                                    Catch e As Exception When FatalError.ReportUnlessCanceled(e, cancellationToken)
```
Comment (CyrusNajmabadi) [inline src/Compilers/VisualBasic/Portable/Symbols/Source/SourceModuleSymbol.vb:706]: these are not clear if they're correct.  can you not change our exception fitlers (unless you totally verify that the new pattern is absolutely teh one we want).

Comment (CyrusNajmabadi) [reply src/Compilers/VisualBasic/Portable/Symbols/Source/SourceModuleSymbol.vb:706]: actually, this is definitely a change in semantics and worries me.  don't do these without talking to @sharwell @tmat @jasonmalinowski  to make sure they're ok.

File: src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs [hunk 2]
```diff
@@ -1188,7 +1188,7 @@ private async Task ProcessCompilationEventsAsync(AnalysisScope analysisScope, An
                     await ProcessEventAsync(completedEvent, analysisScope, analysisStateOpt, cancellationToken).ConfigureAwait(false);
                 }
             }
-            catch (Exception e) when (FatalError.ReportUnlessCanceled(e))
+            catch (Exception e) when (FatalError.ReportUnlessCanceled(e, cancellationToken))
```
Comment (mavasani) [inline src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs:1191]: Seems like passing the `cancellationToken` to `ReportUnlessCanceled` will cause a semantic change. It now does an additional check `cancellationToken.IsCancellationRequested`, which did not happen previously. You may want to revert all changes to `ReportUnlessCanceled` callsites?

Comment (jmarolf) [reply src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs:1191]: yeah, I convinced myself that this was the right thing to do. I'll revert it here and create a separate PR where we can discuss if this change is what we want or not.

File: src/Compilers/VisualBasic/Portable/Symbols/Source/SourceModuleSymbol.vb (hunk 1 above)
Comment (jmarolf) [reply src/Compilers/VisualBasic/Portable/Symbols/Source/SourceModuleSymbol.vb]: reverted

File: src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs (hunk 2 above)
Comment (jmarolf) [reply src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs]: reverted

File: src/Compilers/VisualBasic/Portable/Symbols/Source/SourceModuleSymbol.vb (hunk 1 above)
Comment (tmat) [reply src/Compilers/VisualBasic/Portable/Symbols/Source/SourceModuleSymbol.vb]: Yes, we should not change it at this point.

**Pull request merged**